src/components/MyProfile.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import ProfileRockets from './ProfileRockets';
+import ProfileMission from './ProfileMission';
+import './MyProfile.css';
+
+function MyProfile() {
+  const rockets = useSelector((state) => state.rockets);
+  const missions = useSelector((state) => state.missions);
+
+  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
+  const reservedMissions = missions.filter((mission) => mission.reserved === true);
+
+  const renderList = reservedMissions.map((mission) => (
+    <ProfileMission key={mission.mission_id} content={mission.mission_name} />
+  ));
+
+  const renderReservedRockets = reservedRockets.map((rocket) => (
+    <ProfileRockets
+      key={rocket.id}
+      name={rocket.name}
+    />
+  ));
+
+  return (
+    <div className="queryresult">
+      <div className="Box">
+        <h2 className="Cat-Title">My Missions</h2>
+        <table className="Mission-ProfileTable">
+          <tbody>
+            {renderList}
+          </tbody>
+        </table>
+      </div>
+      <div className="Box">
+        <h2 className="Cat-Title">My Rockets</h2>
+        <table className="Mission-ProfileTable">
+          <tbody>
+            {renderReservedRockets}
+          </tbody>
+        </table>
+      </div>
+    </div>
+  );
+}
+
+export default MyProfile;
```

Comment:
[OPTIONAL] The My Profile section can be enhanced by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (no missions joined or no rockets reserved).

[OPTIONAL] The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.

[OPTIONAL] The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.